lib/cache/operations.js

```diff
@@ -3,11 +3,28 @@
  */
```

Comment:
I think we should move this file out of the `cache` dir and into something like `operationWriter.js` at the top level. We want to keep removing things from the cache dir until it's empty and we can remove it entirely.